src/main/java/org/apache/maven/plugins/shade/filter/SimpleFilter.java

```diff
@@ -42,16 +42,19 @@
 
     private Set<String> excludes;
 
+    private boolean excludeDefaults = true;
+
     /**
      * @param jars set of {@link File}s.
      * @param includes set of includes.
      * @param excludes set of excludes
      */
-    public SimpleFilter( Set<File> jars, Set<String> includes, Set<String> excludes )
+    public SimpleFilter( Set<File> jars, Set<String> includes, Set<String> excludes, boolean excludeDefaults )
```

Comment:
It looks to me we're at a point where we need a second constructor, using the `ArchiveFilter` as argument instead of includes/excludes/excludeDefaults.

Reply:
That would be a refactoring, not just a bug fix. Are you sure you want to me to do that in the same PR?

Reply:
IMHO this is more an improvement than a bugfix. And I'm not sure if others use this SimpleFilter as superclass, so better to keep the original constructor anyway. I don't like the idea that every new field for ArchiveFilter  would translate to an extra constructor, just pass the original instead.

Reply:
Implemented additional constructor as you requested, see https://github.com/apache/maven-shade-plugin/pull/19/commits/975615b596d7a99dd5ee9999782aa0b62400a2dd.